#VRML_SIM R2025a utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/floors/protos/RectangleArena.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/apartment_structure/protos/Wall.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/appearances/protos/BrushedAluminium.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/appearances/protos/Roughcast.proto"

WorldInfo {
}
Viewpoint {
  orientation -0.36375496700451254 0.4800085795124893 0.798294486749062 1.5187817952129992
  position -2.5028297957576067 -5.327325193785998 6.571693281007842
}
TexturedBackground {
}
TexturedBackgroundLight {
}
DirectionalLight {
  direction 0.3 -0.5 -0.7
  intensity 1.5
}
RectangleArena {
  floorSize 3.048 3.048
  floorAppearance PBRAppearance {
    baseColor 0 0 0
    roughness 1
    metalness 0
  }
  wallHeight 0.6096
  wallAppearance BrushedAluminium {
    IBLStrength 0
  }
}
Wall {
  translation 1.52 -0.78 0
  name "wall(2)"
  size 0.01 0.4 0.4
  appearance Roughcast {
    colorOverride 1 0 0
    textureTransform TextureTransform {
      scale 2.4 1
    }
  }
}
Wall {
  translation -0.16 -1.28 0
  rotation 0 0 1 1.5708
  name "wall(8)"
  size 0.1 0.375 0.125
}
Wall {
  translation -0.87 0.19 0
  name "wall(9)"
  size 0.125 0.375 0.105
}
Wall {
  translation 0.97 -0.1 0
  name "wall(12)"
  size 0.125 0.375 0.105
}
Wall {
  translation 0.84 -1.18 0
  rotation 0 0 1 -1.5707953071795862
  name "wall(10)"
  size 0.1 0.375 0.125
}
Wall {
  translation 1.52 -0.78 0
  rotation 0 1 0 -1.5707953071795862
  name "wall(7)"
  size 0.001 0.4 0.4
  appearance Roughcast {
    colorOverride 1 0 0
    textureTransform TextureTransform {
      scale 2.4 1
    }
  }
}
Wall {
  translation 1.52 0.61 0
  name "wall(3)"
  size 0.01 0.4 0.4
  appearance Roughcast {
    colorOverride 0 1 1
    textureTransform TextureTransform {
      scale 2.4 1
    }
  }
}
Wall {
  translation 1.52 0.61 0
  rotation 0 1 0 -1.5707953071795862
  name "wall(6)"
  size 0.001 0.4 0.4
  appearance Roughcast {
    colorOverride 0 1 1
    textureTransform TextureTransform {
      scale 2.4 1
    }
  }
}
Wall {
  translation 0.26 1.31 1.28473e-06
  rotation 0 1 0 -1.5707953071795862
  name "wall(11)"
  size 0.001 0.4 0.4
}
Wall {
  translation -1.52 -0.78 0
  size 0.01 0.4 0.4
  appearance Roughcast {
    colorOverride 0 1 0
    textureTransform TextureTransform {
      scale 2.4 1
    }
  }
}
Wall {
  translation -1.52 -0.78 0
  rotation 0 1 0 1.5708
  name "wall(5)"
  size 0.001 0.4 0.4
  appearance Roughcast {
    colorOverride 0 1 0
    textureTransform TextureTransform {
      scale 2.4 1
    }
  }
}
Wall {
  translation -1.52 0.61 0
  name "wall(1)"
  size 0.01 0.4 0.4
  appearance Roughcast {
    colorOverride 1 1 0
    textureTransform TextureTransform {
      scale 2.4 1
    }
  }
}
Wall {
  translation -1.52 0.61 0
  rotation 0 1 0 1.5708
  name "wall(4)"
  size 0.001 0.4 0.4
  appearance Roughcast {
    colorOverride 1 1 0
    textureTransform TextureTransform {
      scale 2.4 1
    }
  }
}
Solid {
  translation 0.3160178830336076 -0.9560296309371286 0.03733445625002484
  rotation 1.8516739878241456e-17 7.401894485344136e-17 -1 1.4707225602854175
  children [
    DEF rbox Shape {
      appearance PBRAppearance {
        baseColor 1 0 0
        roughness 0.5
        metalness 0
      }
      geometry Box {
        size 0.075 0.075 0.075
      }
    }
  ]
  name "red_box"
  boundingObject USE rbox
  physics Physics {
  }
  linearVelocity -3.6504715410946146e-17 1.9372482962627377e-18 -1.9416027633403226e-13
  angularVelocity -1.6685247315480724e-48 1.3868174713785388e-30 1.0013631615933281e-16
}
Solid {
  translation -0.6698755787164241 -0.23092883593636268 0.03733445625002482
  rotation 9.352785442673541e-18 -2.921475506772778e-17 1 1.77791030204048
  children [
    DEF gbox Shape {
      appearance PBRAppearance {
        baseColor 0 1 0
        roughness 0.5
        metalness 0
      }
      geometry Box {
        size 0.075 0.075 0.075
      }
    }
  ]
  name "green_box"
  boundingObject USE gbox
  physics Physics {
  }
  linearVelocity -1.495581320359337e-16 -1.9786730978231755e-17 -1.9409290635614697e-13
  angularVelocity 1.68424944772765e-15 -3.3684988954552875e-15 -2.980761114493398e-17
}
Solid {
  translation -0.19552489173182752 -1.1056102165180632 0.03733445625002485
  rotation 0.9999132912430966 0.01316852290116786 7.917903772674631e-17 3.141592653589793
  children [
    DEF wbox Shape {
      appearance PBRAppearance {
        roughness 0.5
        metalness 0
      }
      geometry Box {
        size 0.075 0.075 0.075
      }
    }
  ]
  name "white_box"
  boundingObject USE wbox
  physics Physics {
  }
  linearVelocity -1.3618191301336743e-17 1.6809221953827987e-17 -1.9416027633401896e-13
  angularVelocity -1.6842494477276461e-15 -1.1228329651517654e-15 -2.7740072360739894e-16
}
Solid {
  translation -0.727662937946533 0.7329555131833713 0.03733445627518225
  rotation -9.41791805802998e-13 -9.144564205030852e-13 1 1.2921498387510837
  children [
    DEF ybox Shape {
      appearance PBRAppearance {
        baseColor 1 1 0
        roughness 0.5
        metalness 0
      }
      geometry Box {
        size 0.075 0.075 0.075
      }
    }
  ]
  name "yellow_box"
  boundingObject USE ybox
  physics Physics {
  }
  linearVelocity 4.558798775779028e-13 -7.064866434680959e-14 -1.9693942266276085e-10
  angularVelocity 1.8835522990422227e-12 1.215635109721593e-11 -4.453285724316054e-16
}
Solid {
  translation 0.3538308794342245 0.3055841423136196 0.03733445625002483
  rotation 1.9479862509999313e-16 -1.787485486431722e-17 -1 0.4663911557286903
  children [
    DEF bbox Shape {
      appearance PBRAppearance {
        baseColor 0 1 1
        roughness 0.5
        metalness 0
      }
      geometry Box {
        size 0.075 0.075 0.075
      }
    }
  ]
  name "cyan_box"
  boundingObject USE bbox
  physics Physics {
  }
  linearVelocity 2.6968623311561702e-17 -2.739719874065472e-18 -1.941602763340356e-13
  angularVelocity 2.8070824128794036e-16 5.61416482575882e-16 1.0739458002605545e-17
}
DEF BBAlr8 Robot {
  translation -0.4149834494085564 0.6864240955463199 0.055963612823537116
  rotation 0.003956944952207054 -0.004688793335033464 -0.9999811787247331 1.6469486211124766
  children [
    Display {
      children [
        DEF disp Shape {
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      boundingObject USE disp
      physics Physics {
      }
      width 640
      height 480
    }
    DEF gyro Gyro {
      children [
        DEF gyroS Shape {
          geometry Sphere {
            radius 0.01
          }
        }
      ]
      boundingObject USE gyroS
      physics Physics {
      }
    }
    Solid {
      translation 0.11 -0.2 0
      children [
        DEF leftArm Shape {
          appearance PBRAppearance {
            baseColor 0 0 0
          }
          geometry Box {
            size 0.02 0.15 0.09
          }
        }
      ]
      name "solid(2)"
      boundingObject USE leftArm
      physics Physics {
      }
    }
    Solid {
      translation -0.11 -0.2 0
      children [
        DEF rightArm Shape {
          appearance PBRAppearance {
            baseColor 0 0 0
          }
          geometry Box {
            size 0.02 0.15 0.09
          }
        }
      ]
      name "solid(3)"
      boundingObject USE rightArm
      physics Physics {
      }
    }
    DEF frontCam Camera {
      translation -0.00497507 -0.0650495 0.090495
      rotation 0 0 -1 1.5708
      children [
        DEF fcam Shape {
          appearance PBRAppearance {
            baseColor 1 0 0
          }
          geometry Capsule {
            height 0.012
            radius 0.04
          }
          castShadows FALSE
        }
      ]
      name "fcam"
      boundingObject USE fcam
      physics Physics {
      }
      fieldOfView 1
      width 640
      height 480
    }
    DEF castorFront Solid {
      translation 0 -0.15 -0.051
      children [
        DEF castor Shape {
          appearance PBRAppearance {
          }
          geometry Sphere {
            radius 0.005
          }
          castShadows FALSE
        }
      ]
      boundingObject USE castor
      physics Physics {
      }
    }
    DEF castorBack Solid {
      translation 0 0.149 -0.051
      children [
        DEF castor Shape {
          appearance PBRAppearance {
          }
          geometry Sphere {
            radius 0.005
          }
          castShadows FALSE
        }
      ]
      name "solid(1)"
      boundingObject USE castor
      physics Physics {
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 2738.3686440136025
        anchor 0.181 0 -0.03
      }
      device [
        PositionSensor {
          name "leftEncoder"
        }
        RotationalMotor {
          name "leftMotor"
          maxVelocity 100
          maxTorque 100
        }
      ]
      endPoint Solid {
        translation 0.18160993891107124 -5.942596722526775e-06 -0.029997761926176255
        rotation 0.7029196975126059 -0.11869211242808328 0.7012960011978865 3.394296320505423
        children [
          DEF leftWheel Shape {
            appearance PBRAppearance {
              baseColorMap ImageTexture {
                url [
                  "../assets/wheel.jpg"
                ]
              }
            }
            geometry Cylinder {
              height 0.01
              radius 0.03
            }
            castShadows FALSE
          }
        ]
        name "leftWheel"
        boundingObject USE leftWheel
        physics Physics {
        }
        linearVelocity 6.249023706695904e-12 -9.929115892466711e-11 -5.025661940657399e-16
        angularVelocity 2.0342111720780793e-10 2.837127627420983e-09 -3.2997642617632407e-09
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 2748.6810297262846
        anchor -0.181 0 -0.03
      }
      device [
        DEF rightEncoder PositionSensor {
          name "rightEncoder"
        }
        DEF rightMotor RotationalMotor {
          name "rightMotor"
          maxVelocity 100
          maxTorque 100
        }
      ]
      endPoint Solid {
        translation -0.1815837872808982 -9.778717401450976e-05 -0.029850359090241887
        rotation -0.7083799330346715 -0.07767220508067892 -0.7015446522009124 3.286768029957424
        children [
          DEF rightWheel Shape {
            appearance PBRAppearance {
              baseColorMap ImageTexture {
                url [
                  "../assets/wheel.jpg"
                ]
                filtering 5
              }
              textureTransform TextureTransform {
              }
            }
            geometry Cylinder {
              height 0.01
              radius 0.03
            }
            castShadows FALSE
          }
        ]
        name "rightWheel"
        boundingObject USE rightWheel
        physics Physics {
        }
        linearVelocity 1.208020733871104e-09 -1.906375445585271e-10 7.991439389765913e-14
        angularVelocity 3.677073305064514e-09 4.83850398929446e-08 -3.3188234685158596e-09
      }
    }
    DEF body Shape {
      appearance PBRAppearance {
        metalness 0.5
      }
      geometry Cylinder {
        height 0.1
        radius 0.177
      }
      castShadows FALSE
    }
  ]
  boundingObject USE body
  physics Physics {
    centerOfMass [
      0 0 0
    ]
  }
  controller "bbAlr8"
  cpuConsumption 20
  linearVelocity 6.076904473983375e-10 -1.03853863923991e-10 -2.2157456513880976e-12
  angularVelocity -1.1933387509885592e-11 1.4687192325432096e-11 -3.2985831480769016e-09
}
